File: front-end/src/app/services/task.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { TaskInsertReqDto } from '@dto/task/task-insert.req.dto';
import { Observable } from 'rxjs';
import { InsertResDto } from '@dto/InsertResDto';
import { BASE_URL } from '../constant/api.constant';
import { TaskResDto } from '@dto/task/task.res.dto';
import { QuestionResDto } from '@dto/task/question.res.dto';

@Injectable({
  providedIn: 'root',
})
export class TaskService {
  constructor(private base: BaseService) {}

  insert(data: TaskInsertReqDto[]): Observable<InsertResDto> {
    return this.base.post(`${BASE_URL}/task`, data);
  }

  getByLearningId(id: number): Observable<TaskResDto[]> {
    return this.base.get<TaskResDto[]>(`${BASE_URL}/task/?learningId=${id}`);
  }

  getById(id: number): Observable<TaskResDto> {
    return this.base.get(`${BASE_URL}/task/detail?id=${id}`);
  }
}

File: back-end/src/main/java/com/ocire/lms/service/TaskService.java
package com.ocire.lms.service;

import java.util.List;

import com.ocire.lms.dto.InsertResDto;
import com.ocire.lms.dto.task.TaskInsertReqDto;
import com.ocire.lms.dto.task.TaskResDto;
import com.ocire.lms.model.Task;

public interface TaskService {
	List<TaskResDto> getTaskByLearningId(Long learningId);
	
	TaskResDto getById(Long id);
	
	InsertResDto insertTask(List<TaskInsertReqDto> data);
}
